### DISCUSSION TITLE
Errore url interno

### DISCUSSION THREAD

**Discussion by claucas84** [indomus/forum#3086] [HUB: Home Assistant]
Salve, sono un neofita e vi chiedo un po' di pazienza. Ho installato duckdns su home assistant, che ora è raggiungibile solo in https. Ho poi disinstallato duckdns perché non ne ho più bisogno, ma HA è sempre raggiungibile solo da https. Ho letto in varie guide che è normale, la mia domanda è: non avendo più bisogno di esporlo, come faccio a far "tornare" HA in http così da poterlo usare con l'app per smartphone? Grazie

**Comment by snakuzzo** [comment]
Se non hai bisogno di esporlo in https, vuol dire che da fuori casa non vuoi raggiungerlo (salvo tua VPN o simile). È corretto? Vuoi questo? Vuoi raggiungerlo solo dalla LAN di casa tua?
In tal caso ti basterà rimuovere il blocco `http:` nel `contiguration.yaml` e riavviare HA.

**Reply by claucas84** [reply]
Ciao, in realtà sto provando ad esporlo ma senza successo, non riesco ad aprire porte sul router (uso openwrt) nonostante io abbia un indirizzo statico (ISP Eolo). Ora proverò con wireguard, nel frattempo mi fa comodo rimetterlo in http così da poterlo raggiungere con l'app. Grazie per l'aiuto, sei stato gentilissimo (oltre che competente)

**Reply by snakuzzo** [reply]
Cosa c'entra wireguard con l'esposizione su internet?
Eolo ti fornisce un ip pubblico oppure sei nattato?

**Reply by claucas84** [reply]
Eolo mi fornisce un ip pubblico statico (a pagamento). Ho visto che le uniche porte aperte sono la 80 e la 443 (aperte non da me, non ho fatto nessuna configurazione sul router). Con wireguard vorrei mettere su una VPN che mi permetta di accedere da remoto

**Reply by snakuzzo** [reply]
Avresti comunque il discorso dell'apertura porte :)
Comunque ti stai incartando su una cavolata (l'apertura porte appunto).
A ogni modo, se posso darti un consiglio, investi pochi euro/anno, compra un dominio ed esponi con il tunnel gratuito di cloudflare...ne guadagni in sicurezza sicuramente e non dovrai nemmeno aprire porte, ovviamente.
Sul sito abbiamo tutte le guide.

**Reply by claucas84** [reply]
Vado subito a cercare i link per le guide, grazie mille

**Reply by trebestie** [reply]
Se puoi aprire una qualsiasi porta UDP sul router, Wireguard funziona più che bene e su OpenWRT è facile da configurare

**Reply by snakuzzo** [reply]
Non gli serve

**Reply by claucas84** [reply]
In realtà mi servirebbe ma come ho detto prima non riesco ad aprire porte con openwrt. Ho la 80 e la 443 aperte (non da me) e basta. Se volete vi posto la configurazione del router per vedere se c'è qualcosa di sbagliato

**Reply by trebestie** [reply]
Per configurare wireguard sul router openwrt segui le istruzioni contenute in questo [video](https://www.youtube.com/watch?v=E_wZp7t4VaI)

**Reply by claucas84** [reply]
Allora ce l'ho fatta finalmente, sono connesso con la VPN e il tutto funge. L'unico problema è che l'ora dell'handshake non è giusta 
![image](https://github.com/indomus/forum/assets/82472311/97cc7fd8-8b8e-4f74-9630-e1bc83a66fba)
l'ultimo handshake lo segna un'ora fa, ma in realtà mi sono connesso da 5 minuti. Ho fatto delle prove prima di questa connessione (prove non andate a buon fine), l'handshake si verifica ogni volta che mi connetto o solo una volta ogni x secondi/minuti/ore?

**Reply by trebestie** [reply]
Per curiosità sono andato a [leggere](https://www.wireguard.com/protocol/) le specifiche del protocollo. Se ho capito bene la risposta è 'dipende'. 
In ogni caso credo la parte rilevante sia 
`There is a clever pulse mechanism to ensure that the latest keys and handshakes are up to date, renegotiating when needed, by automatically detecting when handshakes are out of date. It uses a separate packet queue per host, so that it can minimize packet loss during handshakes while providing steady performance for all clients.`
E soprattutto
`In other words, you bring the device up, and everything else is handled for you automatically. You don't need to worry about asking it to reconnect or disconnect or reinitialize, or anything of that nature.`

Anche il mio orario è diverso, finora non mi ha dato problemi, quindi non mi preoccuperei.
